server/src/main/java/org/elasticsearch/cluster/metadata/DataLifecycleRolloverMetadata.java

```diff
@@ -0,0 +1,150 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.cluster.metadata;
+
+import org.elasticsearch.cluster.SimpleDiffable;
+import org.elasticsearch.common.Strings;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.common.unit.ByteSizeValue;
+import org.elasticsearch.core.Nullable;
+import org.elasticsearch.core.TimeValue;
+import org.elasticsearch.xcontent.ConstructingObjectParser;
+import org.elasticsearch.xcontent.ObjectParser;
+import org.elasticsearch.xcontent.ParseField;
+import org.elasticsearch.xcontent.ToXContentFragment;
+import org.elasticsearch.xcontent.XContentBuilder;
+import org.elasticsearch.xcontent.XContentParser;
+
+import java.io.IOException;
+import java.util.Objects;
+
+/**
+ * Holds the rollover metadata that configure how data streams managed by Data Lifecycle Management get rolled over.
+ */
+public class DataLifecycleRolloverMetadata implements SimpleDiffable<DataLifecycleMetadata>, ToXContentFragment {
```

Comment:
I don't think we need  `*Metadata` object for rollover? The rollover configuration will be persisted in the `cluster.dlm.default.rollover` cluster setting. We don't need additional persistence for it.

What if we have a `DataLifecycle` object that is a `SimpleDiffable` and a `ToXContentObject` ? It _stores_ only the data retention (similar to what you have in `DataLifecycleMetadata`).
Maybe it has a `getRollover(Settings)` method that returns a `RolloverConfig` by parsing `cluster.dlm.default.rollover` from the Settings. 

We could have another `toXContent` method that also takes an `includeDefaults` parameter  to control if the rollover configuration is included in the output or not. 

The `DataLifecycle` will eventually be part of a `org.elasticsearch.cluster.metadata.DataStream` when configured, and also part of the `org.elasticsearch.cluster.metadata.Template` (that's currently already part of `ComponentTemplate` and `ComposableIndexTemplate`). 

Some of this is potentially just some (re)naming suggestions. What do you think?

Reply:
A recap of what was discussed offline. 

Switched `DataLifecycleMetadata` to `DataLifecycle`, since we will place the `DataLifecycle` in the `DataStream` then I think it makes sense to drop the metadata. I was trying to follow the convention of `MappingMetadata` and `AliasMetadata` but since `Settings` do not have the `Metadata` postfix, it doesn't seem to be such a strong convention.

The rollover is removed completely and left for the next PR.